pkg/v1/tkg/client/upgrade_cluster.go

```diff
@@ -947,7 +949,7 @@ func isNewVSphereTemplateRequired(machineTemplate *capvv1beta1.VSphereMachineTem
 		return true
 	}
 	// If vm template given is the same as we already have in VSphereMachineTemplate
-	if machineTemplate.Spec.Template.Spec.Template != clusterUpgradeConfig.UpgradeComponentInfo.VSphereVMTemplateName {
```

Comment:
two things:
- (nit) the comment does not match the code any more
- what happens when upgrading existing clusters that to not have this annotation set?

Reply:
> what happens when upgrading existing clusters that to not have this annotation set?
In this case a user presumably intends for an upgrade. For example, existing 1.5 clusters don't have this annotation. When a user performs an upgrade using the 1.6 cli they should be expecting an upgrade. This might trip up some main branch users immediately after this gets added to the tree, but for end users this upgrade should be expected.